kie-soup-commons/src/main/java/org/kie/soup/commons/validation/PortablePreconditions.java

```diff
@@ -145,4 +145,32 @@ public static void checkNullMandatory(final String name,
             throw new IllegalArgumentException("Parameter named '" + name + "' should be null!");
         }
     }
-}
+
+    /**
+     * Assert that this parameter is not null and greater than the nonNullValue.
+     * @param name of the parameter.
+     * @param parameter the parameter value.
+     * @param nonNullValue a non null value for executing the comparison.
+     */
+    public static <T> void checkGreaterThan(final String name, final Comparable<T> parameter, final T nonNullValue) {
+        checkNotNull(name, parameter);
+        checkNotNull("nonNullValue", nonNullValue);
+        if (parameter.compareTo(nonNullValue) <= 0) {
+            throw new IllegalArgumentException("Parameter named '" + name + "' must be greater than '" + nonNullValue + "'!");
```

Comment:
soft suggestion: let's include also the parameter's value into the exception message.